ipn/ipnlocal/local_test.go

```diff
@@ -2173,3 +2173,42 @@ func TestOnTailnetDefaultAutoUpdate(t *testing.T) {
 		})
 	}
 }
+
+func TestTCPHandlerForDst(t *testing.T) {
+	b := newTestBackend(t)
+	ip := netip.MustParseAddr("100.100.103.100")
+
+	tests := []struct {
+		dst         netip.AddrPort
+		wantHandler bool
+	}{
+		{
+			dst:         netip.AddrPortFrom(magicDNSIP, TailFSLocalPort),
+			wantHandler: true,
+		},
+		{
+			dst:         netip.AddrPortFrom(ip, TailFSLocalPort),
+			wantHandler: false,
+		},
+		{
+			dst:         netip.AddrPortFrom(magicDNSIP, 80),
+			wantHandler: true,
+		},
+		{
+			dst:         netip.AddrPortFrom(ip, 80),
+			wantHandler: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(fmt.Sprintf("dst=%s", tt.dst), func(t *testing.T) {
+			src := netip.AddrPortFrom(netip.MustParseAddr("100.100.102.100"), 51234)
+			h, _ := b.TCPHandlerForDst(src, tt.dst)
+			if !tt.wantHandler && h != nil {
```

Comment:
I'm guessing you didn't want to the uber-legible `if tt.wantHandler != (h != nil)` eh? 😂

Reply:
Hehe, I'm just not that clever.